choreolib/src/main/java/com/choreo/lib/ChoreoAutoLoop.java

```diff
@@ -0,0 +1,84 @@
+package com.choreo.lib;
+
+import edu.wpi.first.wpilibj.event.EventLoop;
+import edu.wpi.first.wpilibj2.command.button.RobotModeTriggers;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+
+/**
+ * A loop that represents an autonomous routine. This loop is used to handle autonomous trigger
+ * logic and schedule commands.
+ */
+public class ChoreoAutoLoop {
+  /** The name of the loop */
+  protected String name;
+
+  /** The underlying {@link EventLoop} stuff is bound to and polled */
+  protected final EventLoop loop;
+
+  /** A boolean utilized in {@link #enabled()} to resolve trueness */
+  protected boolean hasBeenPolled = false;
+
+  /**
+   * Creates a new loop with a specific name
+   *
+   * @param name Loop name
+   * @see com.choreo.lib.ChoreoAutoFactory#newLoop(String) Creating a loop from a ChoreoAutoFactory
+   */
+  public ChoreoAutoLoop(String name) {
+    this.loop = new EventLoop();
+    this.name = name;
+  }
+
+  /**
+   * A constructor to be used when inhereting this class to instantiate a custom inner loop
+   *
+   * @param name The name of the loop
+   * @param loop The inner {@link EventLoop}
+   */
+  protected ChoreoAutoLoop(String name, EventLoop loop) {
+    this.loop = loop;
+    this.name = name;
+  }
+
+  /**
+   * Creates a {@link Trigger} that is true while this autonomous loop is being polled.
+   *
+   * @return A {@link Trigger} that is true while this autonomous loop is being polled.
+   */
+  public Trigger enabled() {
```

Comment:
So what happens if you want to stop and restart the autonomous routine without restarting code? The loop.enabled().onTrue() triggers won't fire the second time?